src/gt4py/next/config.py

```diff
@@ -81,3 +81,8 @@ def env_flag_to_bool(name: str, default: bool) -> bool:
 CMAKE_BUILD_TYPE: CMakeBuildType = CMakeBuildType[
     os.environ.get("GT4PY_CMAKE_BUILD_TYPE", "debug" if DEBUG else "release").upper()
 ]
+
+#: Experimental, use at your own risk: assume horizontal dimension has stride 1
+UNSTRUCTURED_HORIZONTAL_STRIDE_1: bool = env_flag_to_bool(
```

Comment:
I would rename this to
```suggestion
UNSTRUCTURED_HORIZONTAL_UNIT_STRIDE: bool = env_flag_to_bool(
```
and read this as: "The horizontal dimension is the one with unit stride".

Probably closer to the final solution and also works for cartesian and cpu memory layout where K has unit stride (right?):
```suggestion
UNIT_STRIDE_DIMS: list[str] = ...
```
I'm fine with the initial proposal though.

Reply:
I added a "HAS" to the name. I think it's fine to be very verbose as you should not use that accidentally. ;)